app-android/src/main/java/com/ustadmobile/port/android/view/ClazzListFragment.kt

```diff
@@ -140,4 +165,154 @@ class ClazzListFragment(): UstadListViewFragment<Clazz, ClazzWithListDisplayDeta
     }
 
 
+}
+
+@Composable
+private fun ClazzListScreen(
+    uiState: ClazzListUiState = ClazzListUiState(),
+    onClickClazz: (Clazz) -> Unit = {},
+    onClickSort: () -> Unit = {},
+    onClickFilterChip: (MessageIdOption2) -> Unit = {},
+) {
+
+    LazyVerticalGrid(
+        modifier = Modifier.fillMaxSize(),
+
+        // 600 width of the smallest iPad,
+        // subtracted 20 = horizontal padding & space between cards,
+        // half of 580 is 290
+        columns = GridCells.Adaptive(290.dp)
+    ) {
+
+        item(span = { GridItemSpan(maxLineSpan) }) {
+            UstadListSortHeader(
+                modifier = Modifier.defaultItemPadding(),
+                activeSortOrderOption = uiState.activeSortOrderOption,
+                enabled = uiState.fieldsEnabled,
+                onClickSort = onClickSort
+            )
+        }
+
+        item(span = { GridItemSpan(maxLineSpan) }) {
+            UstadListFilterChipsHeader(
+                modifier = Modifier.defaultItemPadding(),
+                filterOptions = uiState.filterOptions,
+                selectedChipId = uiState.selectedChipId,
+                enabled = uiState.fieldsEnabled,
+                onClickFilterChip = onClickFilterChip,
+            )
+        }
+
+        items(
+            uiState.clazzList.size
+        ) { index ->
+            ClazzListItem(
+                clazz = uiState.clazzList[index],
+                onClickClazz = onClickClazz
+            )
+        }
+    }
+}
+
+@Composable
+fun ClazzListItem(
+    clazz: ClazzWithListDisplayDetails,
+    onClickClazz: (Clazz) -> Unit
+){
+
+    val role = (RoleConstants.ROLE_MESSAGE_IDS.find {
+        it.value == clazz.clazzActiveEnrolment?.clazzEnrolmentRole
+    }?.messageId ?: MessageID.student)
+
+    Card(
+        modifier = Modifier
+            .defaultItemPadding()
+            .clickable {
+                onClickClazz(clazz)
+            },
+        elevation = 10.dp
+    ) {
+        Column(
+            modifier = Modifier.padding(5.dp)
+        ){
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceBetween
+            ) {
+                Column {
+                    Text(clazz.clazzName ?: "")
+                    Text(clazz.clazzDesc ?: "")
+                }
+
+                Row(
+                    horizontalArrangement = Arrangement.spacedBy(5.dp),
+                    verticalAlignment = Alignment.CenterVertically
+                ) {
+                    Icon(
+                        imageVector = Icons.Filled.Badge,
+                        contentDescription = "",
+                    )
+                    Text(messageIdResource(id = role))
+                }
+            }
+
+            Row(
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Icon(
+                    imageVector = Icons.Filled.Lens,
```

Comment:
We need to set the color according to the attendance percentage. Can you make a function called colorForAttendance ?.